enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -270,10 +270,13 @@ class AdjointGenerator
         IRBuilder<> BuilderZ(newi);
         Value *newip = nullptr;
 
-        bool needShadow = is_value_needed_in_reverse<ValueType::ShadowPtr>(
-            TR, gutils, &I,
-            /*toplevel*/ Mode == DerivativeMode::ReverseModeCombined,
-            oldUnreachable);
+        bool needShadow =
+            Mode == DerivativeMode::ForwardMode
+                ? false
```

Comment:
In retrospect I think the answer here is true if the loaded value is used as a pointer. In that case you need to make sure a shadow load is available in the fwd pass. In any case I'd leave a comment, then try testing this.